src/containers/Profile/ProfileInfo/index.tsx

```diff
@@ -66,17 +91,21 @@ const ProfileInfo: FC<ComponentProps> = ({user}) => {
     );
   };
 
-  return (
+  return apiState.progress === APIProgress.REQ || apiState.progress === APIProgress.INIT ? (
+    <div className="ProfileInfo__loader-container">
+      <Loader />
+    </div>
+  ) : (
     <>
       <div className="ProfileInfo">
```

Comment:
@jamessspanggg instead of putting this inside a single return. Can we follow this?

```
if (loading) return<Loader/>
return (
    <div className="ProfileInfo">....</div>
)
```

where `loading` is your logic for API progress. I think this is much cleaner. Let me know what you think.

Reply:
agreed, good suggestion, let me make the changes

Reply:
Made the changes!